### PR TITLE
Don't parse output without link

### PR DESCRIPTION
**Pull request by duzun**
Prevent parsing output when we know it doesn't contain link (a | form | script).

Right now, I can't run Murphy tests with RocketPath enabled, because it tries to parse output and interferes with tests' output.

### CODE REVIEW

**Comment by bh-chaker** [comment]
Hi,

You should have something like this in your config.php to avoid using RocketPath when it is not needed:

  if (php_sapi_name() != "cli"){
    require_once($dirs[$current_hostname]['userland'] . 'RocketPath/rocket_path.class.php');
    RocketSled::runnable(RocketPath::runnable('rules.php', APP_BASE_URI, false));
  }

Thanks,
Chaker.

**Comment by duzun** [comment]
My case is  a very speciffic one and your hint solves it, but I think there is still an issue:
Parsing the output should be an option for two reasons:

1) speed - if I output the right links on my HTML, I don't need RocketPath to reparse my output and basically do nothing;

2) flexibility - what if I want to serve HTML on http://example.com/content.html and JSON on http://example.com/content.json and plain text on http://example.com/content.txt etc ? 
Basically it is the same controller, but different output formats depending on some other conditions.
GitHub uses this aproach and it is a very convenient one, but with RocketPath I just can't do this stuff.

I know, my changes aren't perfect, just a quick patch. The way to go would be to make an option for enabling/disabling output parsing.

**Comment by bh-chaker** [comment]
1) Yes, you are correct. The line 274 'ob_start('rocket_path_parse_output_buffer');' should be optional.

2) You can have the following rule:
'content.(.+)' => array('RunnableClass', 'content_type',),
--> Then you can output the correct format based on $_GET['content_type']

I agree with you on the fact that automatic output parsing should be enabled by default. It will be better to have it in a separate function and call it only when you won't manually call the parse_template function. You can create a pull request to move the line 'ob_start('rocket_path_parse_output_buffer');' to a separate function.
